### DISCUSSION TITLE
Sideband - Settings Import & Simplified UI

### DISCUSSION THREAD

**Discussion by simplyequipped** [markqvist/Reticulum#570] [Sideband]
This is a bit of a brain dump, apologies for that.

Are there any thoughts on adding the ability to select a settings file to import from the Sideband UI? I am thinking about Android usage for less technical folks. I really like the ability to import settings, and being able to share and/or pre-configure some settings on Android would be useful when helping others get started with Sideband.

Maybe this should be handled a little different than the `--import-settings` option, to handle a subset of settings? For example, using `sideband.config.update(imported)` when importing from the UI. I am primarily thinking of not overwriting an existing display name, but could be useful in other circumstances.

Looking for some feedback on this concept, including other possible solutions.

An adjacent discussion is the addition of a Simplified / Advanced UI setting toggle. Distributing Sideband and RNodes would be an ideal communication solution in disaster areas without other infrastructure, like recent Hurricane Helene disaster in the eastern United States.

An initial suggestion would be to hide the following main menu options in Simplified mode:

- Objects & Devices
- Situation Map
- Telemetry
- Hardware
- Encryption Keys
- Plugins

There are probably other elements that could be hidden in Simplified mode, but I have not spent enough time thinking about the balance between functionality and simplicity. A few additional items to potentially hide in Simplified mode, without sufficient consideration:

**Conversations Screen**
- Ingest Paper Message icon

**Individual Conversation Screens**
- Telemetry icon
- Situation Map icon
- Delivery Method icon

I realize this is may not the intended use case for Sideband, so maybe it does not make sense to add something like this.

**Comment by faragher** [comment]
Solely looking importing settings, it's very important and I see two required input mechanisms: files and QR codes. This helps both the Android and desktop versions be more user friendly. 

The ideal workflow for people on-site is:

- Download App
- Scan QR Code
- You're ready

Putting RNode configurations in a QR code is a very clear use case, and it can be generalized as need be, but here are issues to be overcome:

- Sideband needs to be restarted for new configurations, both automatic and manual restarts may confuse people
- What configuration can be imported remotely? RNode configuration? Keys? Telemetry collectors? Preferred distribution nodes?
- How do we prevent malicious actors from taking advantage of this?
- If we authenticate configurations, how do we designate an authority?
- Do we support multiple configurations to be selected from a list?

There are some wonderful use cases here, but a considerable amount of work to be done. The more detailed a specification we can make, the less time programming will consume. Bearing in mind any UI change is a lot of work.

Regarding simplification / advanced modes, it's generally a good idea, but see above about requirements and effort.

**Reply by RFnexus** [reply]
NFC would be another useful option for mobile devices to share hardware configs but not sure how supported that is with Python + Kivy.

**Comment by LinuxinaBit** [comment]
In general, Sideband is not at all user friendly, though I don’t think a “Simplified mode” is the answer; the UI really just needs a complete redesign, but I don’t think that’s a priority right now.
Hopefully Liam Cottle makes a mobile port of Reticulum MeshChat eventually so I can just recommend that, but I wouldn’t hold your breath :)
